rslib/src/import_export/gather.rs

```diff
@@ -0,0 +1,284 @@
+                normal_mut.config_id = 1;
+            } else {
+                // TODO: scheduling case
+                deck.kind = DeckKind::Normal(NormalDeck {
+                    config_id: 1,
+                    ..Default::default()
+                })
+            }
+        }
+    }
+
+    fn reset_cards(&mut self, col: &Collection) {
+        let mut position = col.get_next_card_position();
+        for card in self.cards.iter_mut() {
+            if card.ctype != CardType::New || card.queue != CardQueue::New {
+                card.due = card.original_position.unwrap_or_else(|| {
+                    position += 1;
+                    position - 1
+                }) as i32;
+            }
+            card.interval = 0;
+            card.ease_factor = 0;
+            card.reps = 0;
+            card.lapses = 0;
+            card.original_deck_id = DeckId(0);
+            card.original_due = 0;
+            card.original_position = None;
+            card.queue = CardQueue::New;
+            card.ctype = CardType::New;
+            card.flags = 0;
+        }
+    }
+}
+
+fn gather_media_paths_from_note(
+    note: &Note,
+    inserter: &mut impl FnMut(&str),
+    svg_getter: &impl Fn(NotetypeId) -> bool,
+) {
+    for field in note.fields() {
+        for media_ref in extract_media_refs(field) {
+            inserter(&media_ref.fname_decoded);
+        }
+
+        for latex in extract_latex(field, svg_getter(note.notetype_id)).1 {
+            inserter(&latex.fname);
+        }
+    }
+}
+
+fn gather_media_paths_from_notetype(notetype: &Notetype, inserter: &mut impl FnMut(&str)) {
+    for name in extract_underscored_css_imports(&notetype.config.css) {
+        inserter(name);
+    }
+    for template in &notetype.templates {
+        for template_side in [&template.config.q_format, &template.config.a_format] {
+            for name in extract_underscored_references(template_side) {
+                inserter(name);
+            }
+        }
+    }
+}
+
+fn svg_getter(notetypes: &[Notetype]) -> impl Fn(NotetypeId) -> bool {
+    let svg_map: HashMap<NotetypeId, bool> = notetypes
+        .iter()
+        .map(|nt| (nt.id, nt.config.latex_svg))
+        .collect();
+    move |nt_id| svg_map.get(&nt_id).copied().unwrap_or_default()
+}
+
+impl Collection {
+    fn gather_decks(&mut self, deck_id: Option<DeckId>) -> Result<Vec<Deck>> {
+        if let Some(did) = deck_id {
+            let deck = self.get_deck(did)?.ok_or(AnkiError::NotFound)?;
+            self.storage
+                .deck_id_with_children(&deck)?
+                .iter()
+                .filter(|did| **did != DeckId(1))
+                .map(|did| self.storage.get_deck(*did)?.ok_or(AnkiError::NotFound))
+                .collect()
+        } else {
+            Ok(self
+                .storage
+                .get_all_decks()?
+                .into_iter()
+                .filter(|deck| deck.id != DeckId(1))
+                .collect())
+        }
```

Comment:
To support the selected notes use-case, we'll need to gather decks based on the identified cards, optionally discarding ones that are are outside the selected deck.